cve_bin_tool/cvedb.py

```diff
@@ -531,11 +525,47 @@ def populate_severity(self, severity_data, cursor, data_source):
             except Exception as e:
                 LOGGER.info(f"Unable to insert data for {data_source} - {e}\n{cve}")
 
+    def populate_cve_metrics(self, severity_data, cursor):
+        insert_cve_metrics = self.INSERT_QUERIES["insert_cve_metrics"]
+        del_cve_range = "DELETE from cve_range where CVE_number=?"
+
+        for cve in severity_data:
+            # Check no None values
+            if not bool(cve.get("score")):
+                LOGGER.debug(f"Update score for {cve['ID']}")
+                cve["score"] = "unknown"
+            if not bool(cve.get("CVSS_version")):
+                LOGGER.debug(f"Update CVSS version for {cve['ID']}")
+                cve["CVSS_version"] = "unknown"
+            if not bool(cve.get("CVSS_vector")):
+                LOGGER.debug(f"Update CVSS Vector for {cve['ID']}")
+                cve["CVSS_vector"] = "unknown"
+
+        for cve in severity_data:
+            try:
+                if cve["CVSS_version"] == 2:
```

Comment:
The value of the metric variable should be obtained from the metric table and should be the index and not the string.

So the code should be metric_value = find_metric(metric)

where find_metric is a function taking the metric string as a parameter. The find_metric function will then find the string in the metric table and return the index or an error value indicating that it hasn't been found.